zstd_stream.go

```diff
@@ -146,6 +147,44 @@ func (w *Writer) Close() error {
 	return nil
 }
 
+// cSize is the recommended size of reader.compressionBuffer. This func and
+// invocation allow for a one-time check for validity.
```

Comment:
Oh clever! It took me a moment to understand what this was. I think it is probably more clear than using an `init()` function, which is always a bit weird.